resources/assets/js/voucher.js

```diff
@@ -0,0 +1,95 @@
+import { getCSRFToken } from './utils';
+import { ready } from './ready';
+
+ready(() => {
+  // Add plus 1 voucher click handler to all plus 1 voucher buttons
+  document.querySelectorAll('[data-voucher-amount][data-voucher-user-id]').forEach((element) => {
+    element.addEventListener('click', handlePlus1VoucherClick);
+  });
+});
+
+/**
+ * Send an async request to increase the number of vouchers issued.
+ *
+ * @param {MouseEvent} event
+ */
+const handlePlus1VoucherClick = async (event) => {
+  event.preventDefault();
+  event.stopPropagation();
+
+  /**
+   * @type {HTMLButtonElement}
+   */
+  const element = event.target;
+  const dataset = element.dataset;
+
+  const amount = Number(dataset.voucherAmount);
+  const userId = Number(dataset.voucherUserId);
+
+  // Block user from multiple inputs
+  element.disabled = true;
+
+  const finish = () => {
+    element.disabled = false;
+  };
```

Comment:
This is in a async function. Using a callback here is a bit weird.
But nothin critical.

Reply:
I would even find it more weird to repeat myself on every function exit point. This way: Just put it here…

Reply:
best practice would be to use another function with all the error handling that returns the json if its valid.
That way you only need to handle it once. But this isn't that bad and works. So just keep it.